Mathlib/Dynamics/Ergodic/Function.lean

```diff
@@ -0,0 +1,100 @@
+/-
+Copyright (c) 2023 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import Mathlib.Dynamics.Ergodic.Ergodic
+import Mathlib.MeasureTheory.Function.AEEqFun
+
+/-!
+# Functions invariant under (quasi)ergodic map
+
+In this file we prove that an a.e. strongly measurable function `g : α → X` that is a.e. invariant
+under a (quasi)ergodic is a.e. equal to a constant. We prove several versions of this statement with
+slightly different measurability assumptions. We also formulate a version for
+`MeasureTheory.AEEqFun` functions with all a.e. equalities replaced with equalities in the quotient
+space.
+-/
+
+open Function Set Filter MeasureTheory Topology TopologicalSpace
+
+variable {α X : Type _} [MeasurableSpace α] {μ : MeasureTheory.Measure α}
+
+/-- Let `f : α → α` be a (quasi)ergodic map. Let `g : α → X` is a null-measurable function from `α`
+to a nonempty space with a countable family of measurable sets separating points of a set `s` such
+that `f x ∈ s` for a.e. `x`. If `g` that is a.e.-invariant under `f`, then `g` is a.e. constant. -/
+theorem QuasiErgodic.ae_eq_const_of_ae_eq_comp_of_ae_range₀ [Nonempty X] [MeasurableSpace X]
+    {s : Set X} [HasCountableSeparatingOn X MeasurableSet s] (h : QuasiErgodic f μ)
```

Comment:
could you mention `{f : α → α} {g : α → X}` explicitly in the theorem? I am always perturbed by the autoimplicit feature.